packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -123,6 +124,18 @@ export default class PickerPlugin implements EditorPickerPluginInterface {
      * @param event PluginEvent object
      */
     public onPluginEvent(event: PluginEvent) {
+        if (event.eventType == PluginEventType.ContentChanged &&
+            event.source == ChangeSource.SetContent && this.dataProvider.onContentChanged) {
+                // Undos and other major changes to document content fire this type of event.
+                // Inform the data provider of the current picker placed elements in the body.
+                const elementsInDocument = this.editor.getDocument()

```

Comment:
Use editor.queryElements(). Otherwise elements outside editor will also be quried.